### PR TITLE
chore(#8150): use workspaces for monorepo

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

medic/cht-core#8150

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:try-out-npm-workspaces/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:try-out-npm-workspaces/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:try-out-npm-workspaces/docker-compose/cht-couchdb-clustered.yml)
 
# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -460,12 +448,7 @@ module.exports = function(grunt) {
         stdio: 'inherit', // enable colors!
       },
       'shared-lib-unit': {
-        cmd: () => {
-          const sharedLibs = getSharedLibDirs();
-          return sharedLibs
-            .map(lib => `echo Testing shared library: ${lib} && (cd shared-libs/${lib} && npm test)`)
-            .join(' && ');
-        },
+        cmd: 'npm test --workspaces --if-present',
```
Comment (garethbowen) [inline Gruntfile.js:451]: Seems to run faster somehow?

**Comment by garethbowen** [comment]
@dianabarsan No hurry on this review as I'm on leave for a week and we shouldn't merge till after 4.2.0 anyway, but thought I'd tidy this up before I head off.

**Review by dianabarsan — APPROVED** [review]
Very cool stuff! 
One thing I noticed was that I had to run `npm ci` at root level in order to get workspace dependencies installed. After that, I tried to run `npm install` for a new dependency within a shared lib, and it just worked.  
Cool stuff. If there are any problems with adding shared libs, we can solve them then, since it's such a rare action. 

Awesome work!

**Comment by garethbowen** [comment]
Too many conflicts... replaced with this PR: https://github.com/medic/cht-core/pull/8268

**Pull request closed without merging**